crates/runtime/src/flight.rs

```diff
@@ -283,6 +274,34 @@ fn handle_datafusion_error(e: DataFusionError) -> Status {
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn handle_sql_datafusion_error(e: DataFusionError) -> Status {
+    match e {
+        DataFusionError::External(ref external_error) => {
+            if let Some(execution_error) =
+                external_error.to_string().strip_prefix("Execution error: ")
+            {
+                return Status::invalid_argument(execution_error.to_string());
+            }
+            Status::invalid_argument(e.to_string())
+        }
+        DataFusionError::ArrowError(ref arrow_error, _) => {
+            // The S3 query error is managed here to avoid
+            // introducing custom downcast logic within Datafusion or Object Store.
+            let error_message = format!("{arrow_error:?}");
+            let s3_error_indicator = "store: \"S3\",";
```

Comment:
Putting S3 specific error handling logic in the Flight server code doesn't make sense.

Reply:
Yeah I've thought about that, the tricky thing here is that neither Listing Table nor Object store is something defined in our codebase, so it's hard to implement a S3 specific matching like how we handle the postgres error in postgresconn or github error in GithubRestClient. Do you have any suggestions in terms of avoiding to put this in flight server (since the flightserver has no knowledge of the S3 connector). Thanks!